pymeasure/instruments/agilent/agilentB1500.py

```diff
@@ -1524,56 +1498,246 @@ class SMUCurrentRanging():
         supported_output_ranges = {
             # in combination with ASU also 8
-            'HRSMU': [0, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19],
+            "HRSMU": [0, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19],
             # in combination with ASU also 8,9,10
-            'MPSMU': [0, 11, 12, 13, 14, 15, 16, 17, 18, 19],
-            'HPSMU': [0, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20],
-            'MCSMU': [0, 15, 16, 17, 18, 19, 20],
-            'HCSMU': [0, 15, 16, 17, 18, 19, 20, 22],
-            'DHCSMU': [0, 15, 16, 17, 18, 19, 20, 21, 23],
-            'HVSMU': [0, 11, 12, 13, 14, 15, 16, 17, 18],
-            'UHCU': [0, 26, 28],
-            'HVMCU': [],
-            'UHVU': []
+            "MPSMU": [0, 11, 12, 13, 14, 15, 16, 17, 18, 19],
+            "HPSMU": [0, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20],
+            "MCSMU": [0, 15, 16, 17, 18, 19, 20],
+            "HCSMU": [0, 15, 16, 17, 18, 19, 20, 22],
+            "DHCSMU": [0, 15, 16, 17, 18, 19, 20, 21, 23],
+            "HVSMU": [0, 11, 12, 13, 14, 15, 16, 17, 18],
+            "UHCU": [0, 26, 28],
+            "HVMCU": [],
+            "UHVU": [],
         }
         supported_meas_ranges = {
             **supported_output_ranges,
             # overwrite output ranges:
-            'HVMCU': [0, 19, 21],
-            'UHVU': [0, 15, 16, 17, 18, 19]
+            "HVMCU": [0, 19, 21],
+            "UHVU": [0, 15, 16, 17, 18, 19],
         }
         supported_output_ranges = supported_output_ranges[smu_type]
         supported_meas_ranges = supported_meas_ranges[smu_type]
 
         ranges = {
-            '1 pA': 8,  # for ASU
-            '10 pA': 9,
-            '100 pA': 10,
-            '1 nA': 11,
-            '10 nA': 12,
-            '100 nA': 13,
-            '1 uA': 14,
-            '10 uA': 15,
-            '100 uA': 16,
-            '1 mA': 17,
-            '10 mA': 18,
-            '100 mA': 19,
-            '1 A': 20,
-            '2 A': 21,
-            '20 A': 22,
-            '40 A': 23,
-            '500 A': 26,
-            '2000 A': 28
+            "1 pA": 8,  # for ASU
+            "10 pA": 9,
+            "100 pA": 10,
+            "1 nA": 11,
+            "10 nA": 12,
+            "100 nA": 13,
+            "1 uA": 14,
+            "10 uA": 15,
+            "100 uA": 16,
+            "1 mA": 17,
+            "10 mA": 18,
+            "100 mA": 19,
+            "1 A": 20,
+            "2 A": 21,
+            "20 A": 22,
+            "40 A": 23,
+            "500 A": 26,
+            "2000 A": 28,
         }
 
         # set range attributes
         self.output = Ranging(supported_output_ranges, ranges)
-        self.meas = Ranging(supported_meas_ranges, ranges,
-                            fixed_ranges=True)
+        self.meas = Ranging(supported_meas_ranges, ranges, fixed_ranges=True)
+
+
+######################################
+# SPGU Setup
+######################################
+
+
+class SPGU(Channel):
```

Comment:
Please add both channel classes to the documentation page (agilentB1500.rst), such that users can see also the API of the channels.